core/Makefile

```diff
@@ -12,15 +12,19 @@ DEPFLAGS = -MT $@ -MMD -MP -MF $(DEPDIR)/$(@:.o=.d)
 $(shell mkdir -p $(DEPDIR) >/dev/null)
 $(shell mkdir -p $(DEPDIR)/utils >/dev/null)
 $(shell mkdir -p $(DEPDIR)/modules >/dev/null)
-$(shell mkdir -p $(DEPDIR)/hooks >/dev/null)
 $(shell mkdir -p $(DEPDIR)/drivers >/dev/null)
+$(shell mkdir -p $(DEPDIR)/hooks >/dev/null)
 
 # e.g., 4.9.3 -> 40903
 CXXVERSION := $(shell $(CXX) -dumpversion | sed -e 's/\.\([0-9][0-9]\)/\1/g' -e 's/\.\([0-9]\)/0\1/g' -e 's/^[0-9]\{3,4\}$$/&00/')
-
 CXXVERSION_4_8_OR_HIGHER := $(shell expr $(CXXVERSION) \>= 40800)
-ifeq "$(CXXVERSION_4_8_OR_HIGHER)" "0"
-  $(error g++ 4.8 or higher is required)
+
+IS_GCC := $(shell expr "$(shell $(CXX) --version)" : "g++")
+
+ifeq "$(IS_GCC)" "3"
+		ifeq "$(CXXVERSION_4_8_OR_HIGHER)" "0"
```

Comment:
For Makefile the tab-width should be 8, thus a single hard TAB character should suffice here